ci(coverage): PR runs fail when cached coverage baseline references deleted source files

## Summary

The CI pipeline fails during the `Generate coverage XML` step when a cached coverage baseline from a previous `main` run references source files that have since been deleted or renamed.

## Observed failure

Run: https://github.com/NVIDIA/nvalchemi-toolkit/actions/runs/24261680902/job/70846919065

```
+ uv run coverage xml --fail-under=0 -o nvalchemi.coverage.xml
No source for code: '/__w/nvalchemi-toolkit/nvalchemi-toolkit/nvalchemi/models/composable.py'; see https://coverage.readthedocs.io/en/7.13.4/messages.html#error-no-source
##[error]Process completed with exit code 1.
```

`nvalchemi/models/composable.py` was removed in the model refactor (#66), but the cached `.coverage.baseline` still references it. `coverage xml` tries to read the source of every file in the combined database and aborts.

## Root cause

In `.github/workflows/ci.yml`:

1. Full runs on `main` save `.coverage.baseline` and `.testmondata` under cache keys `coverage-main-${sha}` / `testmon-main-${sha}` with prefix fallback `coverage-main-` / `testmon-main-`.
2. PR runs restore the most recent baseline regardless of whether the PR's source topology matches.
3. `coverage combine` merges the stale baseline with the PR's delta without validating source-file existence.
4. `coverage xml` / `coverage report` then fails on the first missing file.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -196,6 +196,9 @@ omit = [
 ]
 show_missing = true
 fail_under = 75
+# Warn instead of aborting on missing source files (e.g. stale cached
+# baseline referencing files deleted in the current PR).
+ignore_errors = true
 
 [tool.coverage.xml]
 output = "nvalchemi.coverage.xml"

